### PR TITLE
network: Static routes for IPVLAN with L2PROXY

### PR DESCRIPTION
**Pull request by tomponline**
network: Adds ipvlan static routes for l2proxy mode.

### CODE REVIEW

**Comment by tomponline** [comment]
@brauner once https://github.com/lxc/lxc/pull/2964 is ready then I will rebase this on it, but this commit is ready for review https://github.com/lxc/lxc/pull/2968/commits/2d52dab815f8dd9dc6bc9f60e7c64e3a23ad30e3

File: src/lxc/network.c [hunk 1]
```diff
@@ -2818,47 +2856,108 @@ static int lxc_setup_l2proxy(struct lxc_netdev *netdev) {
...
+	/* If a local-loopback ifindex supplied remove the static route to the lo device. */
+	if (lo_ifindex > 0) {
+		if (lxc_ipv6_dest_del(lo_ifindex, ip, 128) < 0) {
+			errCount++;
+			ERROR("Failed to delete ipv6 dest \"%s\" for network ifindex \"%u\"", bufinet6, lo_ifindex);
+		}
+	}
+
+	/* If link is supplied remove the IP neigh proxy entry for this IP on the device. */
+	if(link[0] != '\0') {
```
Comment (brauner) [inline src/lxc/network.c]: nit: missing space between if(

File: src/lxc/network.c [hunk 2]
```diff
@@ -2818,47 +2856,108 @@ static int lxc_setup_l2proxy(struct lxc_netdev *netdev) {
...
-		inet4dev = cur->elem;
-		if (!inet_ntop(AF_INET, &inet4dev->addr, bufinet4, sizeof(bufinet4))) {
-			err = -1;
-			SYSERROR("Failed to convert IP for l2proxy removal on dev \"%s\"", netdev->link);
-			continue; /* Try to remove any other l2proxy entries */
+	/* Perform IPVLAN specific checks. */
+	if (netdev->type == LXC_NET_IPVLAN) {
+		/* Retrieve local-loopback interface index for use with IPVLAN static routes. */
+		lo_ifindex = if_nametoindex(loDev);
+		if (lo_ifindex == 0) {
```
Comment (brauner) [inline src/lxc/network.c]: Hm, above you use `if (!)` and here `if (== 0)`. Can you please make it consistent? :)

**Comment by tomponline** [comment]
@brauner sorry about those two issues, those are fixed now. Thanks

**Pull request merged**